### PR TITLE
let plugins hook tools into MobileTools menu

### PR DESCRIPTION
**Pull request by michael-dev**
This adds a new event TEMPLATE_MOBILETOOLS_DISPLAY
in order to let plugins add items to mobile view
menu.

### CODE REVIEW

File: inc/template.php [hunk 1]
```diff
@@ -1639,6 +1639,9 @@ function tpl_actiondropdown($empty = '', $button = '&gt;') {
         'user_tools' => array('login', 'register', 'profile', 'admin'),
     );
 
+    $evt = new Doku_Event('TEMPLATE_MOBILETOOLS_DISPLAY', $action_structure);
+    if($evt->advise_before()) $action_structure = $evt->data;
```
Comment (Klap-in) [inline inc/template.php]: Sorry for annoying again with code formatting, but could you use { ... }, and move the statement to the new line? Thanks.

**Comment by michael-dev** [comment]
fixed

File: inc/template.php [hunk 2]
```diff
@@ -1639,6 +1639,11 @@ function tpl_actiondropdown($empty = '', $button = '&gt;') {
         'user_tools' => array('login', 'register', 'profile', 'admin'),
     );
 
+    $evt = new Doku_Event('TEMPLATE_MOBILETOOLS_DISPLAY', $action_structure);
+    if ($evt->advise_before()) {
+        $action_structure = $evt->data;
+    }
```
Comment (splitbrain) [inline inc/template.php]: This if should wrap around the whole output. That way plugins could supress the output and do their own thing.

File: inc/template.php [hunk 3]
```diff
@@ -1639,6 +1639,11 @@ function tpl_actiondropdown($empty = '', $button = '&gt;') {
         'user_tools' => array('login', 'register', 'profile', 'admin'),
     );
 
+    $evt = new Doku_Event('TEMPLATE_MOBILETOOLS_DISPLAY', $action_structure);
+    if ($evt->advise_before()) {
+        $action_structure = $evt->data;
```
Comment (splitbrain) [inline inc/template.php]: $action_structure is passed by reference anyway.  There should be no need to reassign here.

**Comment by splitbrain** [comment]
@michael-dev could you adjust the minor things mentioned above?

**Comment by michael-dev** [comment]
fixed

**Comment by splitbrain** [comment]
Hmm I wonder if this shouldn't be changed to make use of a ``dokuwiki\Form`` object now that we have optgroup support. That way plugins wouldn't need to fiddle with raw HTML but could just add/modify the form object.

**Comment by michael-dev** [comment]
I don't see any use of dokuwiki\Form there right now. It looks like an independent change that might want to go in first.

File: inc/template.php (hunk 3 above)
Comment (michael-dev) [inline inc/template.php]: done

File: inc/template.php (hunk 2 above)
Comment (michael-dev) [inline inc/template.php]: done

**Comment by splitbrain** [comment]
Well, the thing is that changing an event later on is a pain because all plugins using it have to be adjusted. That's why I would like to do it right on the first go. Giving plugin authors a Form instance would definitely be more flexible that having them work on raw HTML.

**Comment by splitbrain** [comment]
This should be obsolete with the new Menu system and MENU_ITEMS_ASSEMBLY event.

**Pull request closed without merging**